Protocol.md

```diff
@@ -0,0 +1,177 @@
+# DLC Protocol
+
+This document describes a protocol for two parties to set up a Discreet Log Contract (DLC).
+
+## Preliminaries
+
+The following protocol assumes that both parties have already agreed on the contract terms (the possible outputs of the contract).
+Negotiation of the contract, or decisions on the underlying asset are thus not part of the exchanged information.
+
+## Description
+
+The two parties exchange a set of three messages, and terminates when the non-initiating party broadcasts the Fund Transaction to the Bitcoin network.
+The following figure gives a brief overview of the protocol:
+
+    +-------+                    +-------+
+    |       |                    |       |
+    |       |---- (1) offer  --->|       |
+    |       |                    |       |
+    |       |<--- (2) accept ----|       |
+    |   A   |                    |   B   |
+    |       |---- (3) sign   --->|       |
+    |       |                    |       |
+    |       |                    |  (4) broadcast fund-tx
+    |       |                    |       |
+    +-------+                    +-------+
+
+### Offer
+
+The initiating party starts the protocol by sending an `offer` message to the other party.
+This message includes the following information:
+1. Contract information (an array of triples defined below)
+1. Oracle information (optional)
+1. A's fund amount (satoshi)
+1. A's public key
+1. A's inputs
+1. A's change outputs (optional)
+1. Estimatesmartfee (satoshi/kbyte)
+1. CET CSV delay (optional)
+1. Refund locktime
+
+#### Contract information
+Contract information consists of a list of triples to be used to create CETs.
+The triple consists of:
+`message` - `a_output` (satoshi) - `b_output` (satoshi)
+where `message` is the data on which the oracle will be providing a (set of) signatures, and `a_output` and `b_output` represent the amount that A and B will receive for this outcome respectively.
+The interpretation of the message semantics is left to the application.
```

Comment:
I think this is fine for now but we should consider standardizing what is signed to being an index or a hash I think (in the future)

Reply:
Yes I agree it needs to be further defined. But maybe it has more to do with some oracle specifications?